### PR TITLE
Detect pseudo/v2

### PR DESCRIPTION
**Pull request by victorjulien**
https://redmine.openinfosecfoundation.org/issues/7002 with a few minor cleanups.

### CODE REVIEW

**Review by inashivb — COMMENTED** [review]
1. Haven't looked much into 6827aa97.
2. Inline comment left about naming.

Q: Are you missing `detect-csum.c` on purpose?

File: src/detect-engine-analyzer.c [hunk 1]
```diff
@@ -977,6 +977,9 @@ void EngineAnalysisRules2(const DetectEngineCtx *de_ctx, const Signature *s)
     if (s->mask & SIG_MASK_REQUIRE_ENGINE_EVENT) {
         jb_append_string(ctx.js, "engine_event");
     }
+    if (s->mask & SIG_MASK_REQUIRE_NEED_REAL_PKT) {
```
[the remark this answers is not in the document]
Comment (victorjulien) [reply src/detect-engine-analyzer.c:980]: Good point, will adjust.

**Comment by victorjulien** [comment]
> Q: Are you missing `detect-csum.c` on purpose?

Nope, I guess I did `git grep` wrong. Will update.

**Comment by suricata-qa** [comment]
ERROR:
 

ERROR: QA failed on ASAN_TLPR1_suri.
  

Pipeline 20443

**Comment by victorjulien** [comment]
```
suricata: detect-stream_size.c:121: int DetectStreamSizeMatch(DetectEngineThreadCtx *, Packet *, const Signature *, const SigMatchCtx *): Assertion `!((((p)->flags & ((1UL << (9))|(1UL << (27))))))' failed.
suricata: detect-stream_size.c:121: int DetectStreamSizeMatch(DetectEngineThreadCtx *, Packet *, const Signature *, const SigMatchCtx *): Assertion `!((((p)->flags & ((1UL << (9))|(1UL << (27))))))' failed.
```

**Comment by victorjulien** [comment]
replaced by #11088

**Pull request closed without merging**